apps/admin/frontend/src/api.ts

```diff
@@ -102,3 +104,243 @@ export const unprogramCard = {
+
+type GetWriteInsProps = QueryProps<'getWriteIns'>;
+export const getWriteIns = {
+  queryKey(props?: GetWriteInsProps): QueryKey {
+    return props ? ['getWriteIns', props] : ['getWriteIns'];
+  },
+  useQuery(props?: GetWriteInsProps) {
+    const apiClient = useApiClient();
+    return useQuery(this.queryKey(props), () => apiClient.getWriteIns(props));
+  },
+} as const;
+
+type GetWriteInSummaryProps = QueryProps<'getWriteInSummary'>;
+export const getWriteInSummary = {
+  queryKey(props?: GetWriteInSummaryProps): QueryKey {
+    return props ? ['getWriteInSummary', props] : ['getWriteInSummary'];
+  },
+  useQuery(props?: GetWriteInSummaryProps) {
+    const apiClient = useApiClient();
+    return useQuery(this.queryKey(props), () =>
+      apiClient.getWriteInSummary(props)
+    );
+  },
+} as const;
+
+type GetWriteInAdjudicationTableProps =
+  QueryProps<'getWriteInAdjudicationTable'>;
+export const getWriteInAdjudicationTable = {
+  queryKey(props?: GetWriteInAdjudicationTableProps): QueryKey {
+    return props
+      ? ['getWriteInAdjudicationTable', props]
+      : ['getWriteInAdjudicationTable'];
+  },
+  useQuery(props: GetWriteInAdjudicationTableProps) {
+    const apiClient = useApiClient();
+    return useQuery(this.queryKey(props), () =>
+      apiClient.getWriteInAdjudicationTable(props)
+    );
+  },
+} as const;
+
+type GetWriteInImageProps = QueryProps<'getWriteInImage'>;
+export const getWriteInImage = {
+  queryKey(props?: GetWriteInImageProps): QueryKey {
+    return props ? ['getWriteInImage', props] : ['getWriteInImage'];
+  },
+  useQuery(props: GetWriteInImageProps) {
+    const apiClient = useApiClient();
+    return useQuery(this.queryKey(props), () =>
+      apiClient.getWriteInImage(props)
+    );
+  },
+} as const;
+
+// Grouped Invalidations
+
+function invalidateCastVoteRecordQueries(queryClient: QueryClient) {
+  void queryClient.invalidateQueries(getCastVoteRecordFileMode.queryKey());
+  void queryClient.invalidateQueries(getCastVoteRecordFiles.queryKey());
+  void queryClient.invalidateQueries(getCastVoteRecords.queryKey());
+}
+
+function invalidateWriteInQueries(queryClient: QueryClient) {
+  void queryClient.invalidateQueries(getWriteIns.queryKey());
+  void queryClient.invalidateQueries(getWriteInSummary.queryKey());
+  void queryClient.invalidateQueries(getWriteInAdjudicationTable.queryKey());
+}
+
+// Mutations
+
+export const configure = {
+  useMutation() {
+    const apiClient = useApiClient();
+    const queryClient = useQueryClient();
+    return useMutation(apiClient.configure, {
+      onSuccess() {
+        void queryClient.invalidateQueries(
+          getCurrentElectionMetadata.queryKey()
+        );
+      },
+    });
+  },
+} as const;
+
+export const unconfigure = {
+  useMutation() {
+    const apiClient = useApiClient();
+    const queryClient = useQueryClient();
+    return useMutation(apiClient.unconfigure, {
+      onSuccess() {
+        // invalidate all queries but auth
```

Comment:
I wonder if we should actually put in some wildcard key to actually invalidate _all_ queries in the cache. It might be hard to keep this up to date as we add more queries.

Reply:
if you just call `invalidateQueries` with no arguments it will do just that. I had that initially, but I was reticent to invalidate the auth status query. Could be fine, or maybe there's a way to invalidate all but.

Reply:
I can't imagine invalidating the auth status query would have much impact. It's refetched constantly anyway.

Reply:
Right now it'll return `null` for the whole app while loading, but that doesn't really matter / could be changed.

Reply:
After invalidating the auth status query, the stale auth status will still be shown while the new status is fetching. That's the foundational contract of react-query (it's a caching strategy called "stale while revalidate")